sdk/storage/storage-blob/src/Clients.ts

```diff
@@ -3197,6 +3200,109 @@ export interface BlockBlobUploadOptions extends CommonOptions {
+   */
+  timeoutInSeconds?: number;
+  /**
+   * An implementation of the `AbortSignalLike` interface to signal the request to cancel the operation.
+   * For example, use the &commat;azure/abort-controller to create an `AbortSignal`.
+   *
+   * @type {AbortSignalLike}
+   * @memberof BlockBlobSyncUploadFromURLOptions
+   */
+  abortSignal?: AbortSignalLike;
+  /**
+   * Optional. Specifies a user-defined name-value pair associated with the blob. If no name-value
+   * pairs are specified, the operation will copy the metadata from the source blob or file to the
+   * destination blob. If one or more name-value pairs are specified, the destination blob is
+   * created with the specified metadata, and metadata is not copied from the source blob or file.
+   * Note that beginning with version 2009-09-19, metadata names must adhere to the naming rules
+   * for C# identifiers. See Naming and Referencing Containers, Blobs, and Metadata for more
+   * information.
+   *
+   * @type {Metadata}
+   * @memberof BlockBlobSyncUploadFromURLOptions
+   */
+  metadata?: Metadata;
+  /**
+   * Optional. Version 2019-07-07 and later.  Specifies the name of the encryption scope to use to
+   * encrypt the data provided in the request. If not specified, encryption is performed with the
+   * default account encryption scope.  For more information, see Encryption at Rest for Azure
+   * Storage Services.
+   *
+   * @type {string}
+   * @memberof BlockBlobSyncUploadFromURLOptions
+   */
+  encryptionScope?: string;
+  /**
+   * Access tier.
+   * More Details - https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers
+   *
+   * @type {BlockBlobTier | string}
+   * @memberof BlockBlobSyncUploadFromURLOptions
+   */
+  tier?: BlockBlobTier | string;
+  /**
+   * Specify the md5 calculated for the range of bytes that must be read from the copy source.
+   * @type {Uint8Array}
+   * @memberof BlockBlobSyncUploadFromURLOptions
+   */
+  sourceContentMD5?: Uint8Array;
+  /**
+   * Blob tags.
+   *
+   * @type {Tags}
+   * @memberof BlockBlobSyncUploadFromURLOptions
+   */
+  tags?: Tags;
+  /**
+   * Optional, default is true.  Indicates if properties from the source blob should be copied.
+   *
+   * @type {boolean}
+   * @memberof BlockBlobSyncUploadFromURLOptions
+   */
+  copySourceBlobProperties?: boolean;
+  /**
+   * HTTP headers to set when uploading to a block blob.
+   *
+   * @type {BlobHTTPHeaders}
+   * @memberof BlockBlobSyncUploadFromURLOptions
+   */
+  blobHTTPHeaders?: BlobHTTPHeaders;
+  /**
+   * Conditions to meet for the destination Azure Blob.
+   *
+   * @type {BlobRequestConditions}
+   * @memberof BlockBlobSyncUploadFromURLOptions
+   */
+  conditions?: BlobRequestConditions;
+  /**
+   * Customer Provided Key Info.
+   *
+   * @type {CpkInfo}
+   * @memberof BlockBlobSyncUploadFromURLOptions
+   */
+  customerProvidedKey?: CpkInfo;
+  /**
+   * Optional. Conditions to meet for the source Azure Blob.
```

Comment:
Source might be not an Azure Blob?

Reply:
The source could be external. But in that case, this option won't apply.
And I have no clue whether this interface applies to Azure File.
What do you suggest we change this to?